lang/cs/Org.Apache.REEF.Tests/Functional/Bridge/TestContextStack.cs

```diff
@@ -0,0 +1,226 @@
+            {
+                _requestor = evaluatorRequestor;
+            }
+
+            public void OnNext(IDriverStarted value)
+            {
+                _requestor.Submit(_requestor.NewBuilder().Build());
+            }
+
+            public void OnNext(IAllocatedEvaluator value)
+            {
+                value.SubmitContext(Common.Context.ContextConfiguration.ConfigurationModule
+                    .Set(Common.Context.ContextConfiguration.Identifier, ContextOneId)
+                    .Build());
+            }
+
+            public void OnNext(IActiveContext value)
+            {
+                Logger.Log(Level.Info, "ContextId: " + value.Id);
+                switch (value.Id)
+                {
+                    case ContextOneId:
+                        var contextConfig =
+                            Common.Context.ContextConfiguration.ConfigurationModule.Set(
+                                Common.Context.ContextConfiguration.Identifier, ContextTwoId)
+                                .Build();
+                        var stackingContextConfig =
+                            TangFactory.GetTang()
+                                .NewConfigurationBuilder()
+                                .BindImplementation(GenericType<IInjectableInterface>.Class,
+                                    GenericType<InjectableInterfaceImpl>.Class)
+                                    .Build();
+
+                        value.SubmitContext(Configurations.Merge(stackingContextConfig, contextConfig));
+                        break;
+                    case ContextTwoId:
+                        value.SubmitTask(
+                            TaskConfiguration.ConfigurationModule.Set(TaskConfiguration.Identifier, "contextStackTestTask")
+                                .Set(TaskConfiguration.Task, GenericType<TestContextStackTask>.Class)
+                                .Build());
+                        break;
+                    default:
+                        throw new Exception("Unexpected ContextId: " + value.Id);
+                }
+            }
+
+            public void OnNext(ICompletedTask value)
+            {
+                Logger.Log(Level.Info, TaskValidationMessage);
+                value.ActiveContext.Dispose();
+            }
+
+            public void OnNext(IClosedContext value)
+            {
+                Logger.Log(Level.Info, ClosedContextValidationMessage);
+                if (value.ParentContext != null)
+                {
+                    value.ParentContext.Dispose();
+                }
+            }
+
+            public void OnNext(IRunningTask value)
+            {
+                Logger.Log(Level.Info, "Test task is running.");
+            }
+
+            public void OnNext(ICompletedEvaluator value)
+            {
+                Logger.Log(Level.Info, "Evaluator completed.");
+            }
+
+            public void OnNext(IFailedTask value)
```

Comment:
Not needed as, it repeats the default behavior. However, it would be better to return this as a test failure to the client, as this represents a successfully failed Driver, which is different from other abnormal exits.